src/ng/urlUtils.js

```diff
@@ -94,3 +94,15 @@ function urlIsSameOrigin(requestUrl) {
   return (parsed.protocol === originUrl.protocol &&
           parsed.host === originUrl.host);
 }
+
+  /**
+  * Parse a request URL & determine if its file protocol to avoid $$cookieReader of XSRF token in case of file.
+  *
+  * @param {string|object} requestUrl The url of the request as a string that will be resolved
+  * or a parsed URL object.
+  * @returns {boolean} Whether the request is for the same origin as the application document.
+  */
+function urlIsFromFile(requestUrl) {
+  var parsed = (isString(requestUrl)) ? urlResolve(requestUrl) : requestUrl;
```

Comment:
Let's avoid parsing the URL twice (once in `urlIsSameOrigin` and one here).
Let's parse it once and pass tge parsed URL to both functions.